src/server/system_services/system_store.js

```diff
@@ -352,6 +357,8 @@ class SystemStore extends EventEmitter {
         this.START_REFRESH_THRESHOLD = 10 * 60 * 1000;
         this.FORCE_REFRESH_THRESHOLD = 60 * 60 * 1000;
         this.SYSTEM_STORE_LOAD_CONCURRENCY = config.SYSTEM_STORE_LOAD_CONCURRENCY || 5;
+        this.source = (process.env.HOSTNAME && process.env.HOSTNAME.indexOf("endpoint") === -1) ? SOURCE.DB : SOURCE.CORE;
```

Comment:
It's better to pass the source from outside rather than infer it inside the constructor. It will give us better control later if needed. 
Another option is to add a const in config.js, which is controlled by an env var. It will allow us to tweak it if any bugs are found in the field.